buildbot/master/files/config/master.cfg

```diff
@@ -145,49 +159,57 @@ def branch_priority(builder, requests):
     return requests[0]
 
 
-class DynamicServoBuilder(util.BuilderConfig):
+class DynamicRepoBuilder(util.BuilderConfig):
     """\
-    Builder which uses DynamicServoFactory to run steps
-    from a YAML file in the main servo repo.
+    Builder which uses DynamicGitUrlFactory to run steps
+    from a YAML file in the repo.
     """
-    def __init__(self, name, slavenames, environment):
+    def __init__(self, repo_owner, repo_name, name, slavenames, environment):
         # util.BuilderConfig is an old-style class so we cannot use super()
         # but must hardcode the superclass here
         util.BuilderConfig.__init__(
             self,
             name=name,
             slavenames=slavenames,
-            factory=factories.DynamicServoFactory(name, environment),
+            factory=factories.DynamicGitUrlFactory("https://github.com/%s/%s" % (repo_owner, repo_name),
+                                                   name, environment,
+                                                   "etc/ci/buildbot_steps.yml"),
```

Comment:
Should we hardcode this path? Webrender will need its own steps.yaml file, and it seems a little silly to add `etc/ci` directories just for that. I'm not opposed to leaving this hardcoded, though.